docs/import/color_function_util.js

```diff
@@ -25,22 +179,136 @@ function withColor(td, layer, property, index) {
       td.append(createColorBox(colorFunction['color']));
       break;
     case ColorStyle.DISCRETE:
-      const colorObject = colorFunction['colors'];
-      const colorSet = new Set();
-      Object.keys(colorObject).forEach((propertyValue) => {
-        const color = colorObject[propertyValue];
-        if (!colorSet.has(color)) {
-          colorSet.add(color);
-          td.append(createColorBox(colorObject[propertyValue]));
-        }
-      });
+      createColorBoxesForDiscrete(colorFunction, td);
       break;
     default:
       setStatus(ILLEGAL_STATE_ERR + 'unrecognized color function: ' + layer);
   }
+  td.on('click', () => onClick(td, colorFunction['current-style']));
   return td;
 }
 
+/**
+ * On click function for color tds - updates the globalTd.
+ * @param {JQuery<HTMLElement>} td
+ * @param {enum} type
+ */
+function onClick(td, type) {
+  if ($(td).hasClass('na')) {
+    return;
+  }
+  const colorFunctionDiv = $('#color-fxn-editor');
+  if (colorFunctionDiv.is(':visible') && td === globalTd) {
+    colorFunctionDiv.hide();
+    return;
+  }
+  colorFunctionDiv.show();
+  if (td === globalTd) {
+    return;
+  }
+  globalTd = td;
+  $('#' + colorStyleTypeStrings.get(type) + '-radio')
+      .prop('checked', true)
+      .trigger('change');
+}
+
+/**
+ * Switches the schema of {@link globalTd} to the given type.
+ * @param {enum} type
+ */
+function switchSchema(type) {
+  const colorFunction = getColorFunction();
+  $('.color-type-div').hide();
+  switch (type) {
+    case ColorStyle.SINGLE:
+      $('#single-color-radio').prop('checked', true);
+      $('#single-color-picker').val(colorFunction['color']);
+      $('#single').show();
+      break;
+    case ColorStyle.CONTINUOUS:
+      $('#continuous-radio').prop('checked', true);
+      $('#continuous-color-picker').val(colorFunction['color']);
+      populatePropertyPicker($('#continuous-property-picker'));
+      $('#continuous').show();
+      break;
+    case ColorStyle.DISCRETE:
+      $('#discrete-radio').prop('checked', true);
+      const propertyPicker = $('#discrete-property-picker');
+      populatePropertyPicker(propertyPicker);
+      if (propertyPicker.val()) populateDiscreteColorPickers();
```

Comment:
Agreed, but sort of nice because it means we don't have to make a bunch of color pickers for discrete if this is a single-color data set. I can push it down into populateDiscreteColorPickers if that would help.

Reply:
As discussed offline, TODO and remove in follow-up?

Reply:
I moved this check down a level inside populateDiscreteColorPickers and I think it should stay there. If the 'field' field isn't set to anything, it doesn't make sense to create any value color pickers right?

Reply:
Sounds good.